msu/systems/debug/debug_mod_addon.nut

```diff
@@ -20,6 +20,16 @@
 		return ::MSU.System.Debug.isFullDebugForMod(this.Mod.getID());
 	}
 
+	function enable()
```

Comment:
As this only sets the Default flag, I'd suggest renaming the function to something like `setDefaultFlag`. Because the name `enable` seems to imply to me what I suggested some time ago i.e. enable/disable the debug system for a mod keeping the flags in their current state.

And I still believe that a Boolean to enable/disable the Debug system while keeping flag values intact is a useful feature, as it allows e.g. a modder to make a setting that can be toggled on/off by a user to enable/disable debug. I don't see how All or None is better than that. A good example can be whereby a Modder creates a "Debug" page in his settings panel where all the different flag types can be toggled on/off. And when a user is facing a certain problem, the modder can simply ask the user to toggle a certain flag on/off while enabling/disabling the Debug system as necessary to prevent log spam.

Reply:
It's supposed to be a very simplified and short helper function. People can read the documentation.

I don't get how having a separate enable and disable toggle helps in your example. If there are already settings for each flag, why do you need that on top to avoid log spam? Just tell the user to enable specific flags.
I don't like the idea of having to watch multiple things for the debugging to be enabled or disabled, not only the flag bool but then also your general enable and perhaps even fulldebug.

Reply:
While I don't like that this change is being done in an unrelated PR, and initially I agreed with Midas, I think doing it this way is fine and honestly the separate boolean is a pretty niche system.
I think having it this way is fine and we just need to release.